WalletWasabi.Fluent/ViewModels/ApplicationViewModel.cs

```diff
@@ -0,0 +1,9 @@
+﻿using WalletWasabi.Fluent.Providers;
+
+namespace WalletWasabi.Fluent.ViewModels
+{
+	public partial class ApplicationViewModel : ViewModelBase, ICanShutdownProvider
+	{
+		bool ICanShutdownProvider.CanShutdown() => Services.WalletManager.AnyCoinJoinInProgress();
```

Comment:
This needs to change in the future, AFAIK `CoinJoinManager` will have a function something like `AnyCoinInCriticalPhase()` which going to tell us when to prevent shutdown.